plugins/bot_manager.py

```diff
@@ -48,6 +48,7 @@ async def reply_exception(ctx: Context) -> None:
 @privileged
 async def load_command(ctx: Context, plugin: PluginConverter) -> None:
     """Load a plugin."""
+    assert manager
```

Comment:
for whatever reason, it appears that "global" assertions do not transfer into these functions to the satisfaction of whichever release of pyright is running on the CI checks. This was the only way I found to successfully satisfy pyright.

Reply:
how about
```py
def get_current_manager() -> plugins.PluginManager:
    manager = plugins.PluginManager.of(__name__)
    assert manager
    return manager

manager = get_current_manager()
```
in the module scope?